packages/cli/src/ui/components/shared/text-buffer.ts

```diff
@@ -249,6 +241,59 @@ export const findWordEndInLine = (line: string, col: number): number | null => {
   return null;
 };
 
+// Initialize segmenter for word boundary detection
+const segmenter = new Intl.Segmenter(undefined, { granularity: 'word' });
+
+function findPrevWordBoundary(line: string, cursorCol: number): number {
+  const codePoints = toCodePoints(line);
+  // Convert cursorCol (CP index) to string index
+  const prefix = codePoints.slice(0, cursorCol).join('');
+  const cursorIdx = prefix.length;
+
+  const segments = Array.from(segmenter.segment(line));
+  let targetIdx = 0;
+
+  // Search backwards for the start of the previous word
+  for (let i = segments.length - 1; i >= 0; i--) {
+    const seg = segments[i];
+    const segStart = seg.index;
+
+    // We are looking for a word start strictly before the cursor
+    if (segStart >= cursorIdx) continue;
```

Comment:
nit: I think `continue` can contribute to misunderstanding the code because it acts sort of like a goto in this context.

Consider inverting the condition and putting the following if statement inside.